### ISSUE TITLE
[tf.learn] MetricSpec doesn't work with streaming_mean

### ISSUE THREAD

**Issue by elibixby** [tensorflow/tensorflow#5287]
MetricSpecs always pass both values and labels to metrics, meaning metrics which don't use labels like streaming_mean are broken.

**Comment by shoyer** [comment]
I wouldn't say that streaming_mean is broken. Rather, it just doesn't fit the mold of the other metrics. We should consider moving it and other similar "metrics" like streaming_concat into their own library of primitives for streaming computation.

**Comment by elibixby** [comment]
Maybe.. To me both the Metric and Monitor patterns feel too restrictive to me, for what they are (basically run this thing in the training loop somewhere). They feel similar but different enough to be frustrating. I don't have any concrete suggestions yet but I'm thinking about it (along with a lot of other tf.learn thoughts).
